SFPortfolioApp/SFPortfolioApp/Components/ScannerView/ScannerView.swift

```diff
@@ -0,0 +1,214 @@
+//
+//  ScannerView.swift
+//  SFPortfolioApp
+//
+//  Created by Stoyko Kolev on 27.10.21.
+//
+
+import SFBaseKit
+import AVFoundation
+
+class ScannerView: BaseView {
+    
+    // MARK: - Properties
+    private var captureSession: AVCaptureSession?
+    private var rearCamera: AVCaptureDevice?
+    private var rearCameraInput: AVCaptureDeviceInput?
+    private var videoPreviewOutput: AVCaptureVideoDataOutput?
+    private var videoPreviewLayer: AVCaptureVideoPreviewLayer?
+    private var viewModel: ScannerViewModelProtocol! {
+        didSet {
+            setupBinding()
+        }
+    }
+    
+    // MARK: - IBOutlets
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+    @IBOutlet private weak var infoLabel: UILabel!
+    @IBOutlet private weak var videoPreviewView: UIView!
+    @IBOutlet private weak var captureView: UIView!
+    @IBOutlet private weak var dimmedView: UIView!
+    @IBOutlet private weak var captureViewHeight: NSLayoutConstraint!
+    
+    // MARK: - Lifecycle
+    override func draw(_ rect: CGRect) {
+        super.draw(rect)
+        addDimmedMask()
+    }
+    
+    deinit {
+        viewModel.stop()
+    }
+    
+    // MARK: - IBActions
+    @IBAction private func didChangeSegmentControlValue(_ sender: UISegmentedControl) {
+        guard let title = sender.titleForSegment(at: sender.selectedSegmentIndex),
+              let scanType = ScanType(rawValue: title) else { return }
+        
+        resizeScanArea(for: scanType)
+        addDimmedMask()
+        viewModel.changeType(scanType)
+    }
+    
+    // MARK: - Public Functions
+    /// Configures scanning preview.
+    func setup() {
+        createSession()
+        configureCamera()
+        addSessionInput()
+        addSessionOutput()
+    }
+    
+    /// Starts scanning.
+    func startScan() {
+        viewModel.start()
+        DispatchQueue.global(qos: .userInteractive).async { [weak self] in
+            guard let session = self?.captureSession,
+                  !session.isRunning else { return }
+            
+            session.startRunning()
+            DispatchQueue.main.async { [weak self] in
```

Comment:
[optional]

No need to weakify self here, as self is already weakly captured by the outer closure.

Reply:
Removed